### PR TITLE
Workaround for Windows Live authentication

### PR DESCRIPTION
**Pull request by mourjan**
A while ago, Microsoft enforced their oauth security by disallowing redirect url to contain any parameters.
Hence the following redirect url does not work:
http://mywebsite.com/path_to_hybridauth/?hauth.done=Live

A workaound created by @tohweesiang was to provide another redirect url format in the case of Live authentication, so the redirect url will become like:
http://mywebsite.com/path_to_hybridauth/live.php

If this pull request is accepted than [Hybridauth's documentation for Windows Live](http://hybridauth.sourceforge.net/userguide/IDProvider_info_Live.html) should be changed as well,

regards

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "hybridauth/hybridauth",
+    "name": "mourjan/hybridauth",
```
Comment (StorytellerCZ) [inline composer.json]: Please fix this back.

Comment (mourjan) [inline composer.json]: Done..

am I supposed to submit another Pull Request?

Bassel

File: hybridauth/Hybrid/Auth.php [hunk 2]
```diff
@@ -352,6 +352,9 @@ public static function logoutAllProviders() {
 	 * @param string $mode PHP|JS
 	 */
 	public static function redirect($url, $mode = "PHP") {
+		if(!$mode){
+			$mode = 'PHP';
```
Comment (christ0ph3r) [inline hybridauth/Hybrid/Auth.php]: I am not sure if this is needed.   Can you explain why it is?

```
<?php 

function redirect($url, $mode = "PHP") {
  echo $mode;
}

redirect('https://test.com/');
?>
```

will output PHP because the second parameter is already set 

$mode = "PHP"

so I do not understand why check it again.

Comment (christ0ph3r) [inline hybridauth/Hybrid/Auth.php]: Nevermind.  I understand now.  So incase its empty this is fallback.

Comment (mourjan) [inline hybridauth/Hybrid/Auth.php]: Exactly, in some cases **$mode** is passed as an empty string, and in this case **$mode** will not be initialized by '_PHP_' value since it is not **null**

File: composer.json (hunk 1 above)
Comment (StorytellerCZ) [inline composer.json]: Just pushing to this branch should do.
